src/PrestaShopBundle/EventListener/ModuleActivatedListener.php

```diff
@@ -111,21 +70,34 @@ public function onKernelController(ControllerEvent $event)
         }
 
         [$controllerObject, $methodName] = $controller;
-        $moduleActivated = $this->getAnnotation($controllerObject, $methodName);
 
-        if (null === $moduleActivated) {
+        $method = new \ReflectionMethod($controllerObject, $methodName);
```

Comment:
I'm just worried about this, since reflection is known to be quite demanding in performance I think it would be nice to have some cache around this attribute fetching process Especially if it's in a listener that is executed on every request for an attribute that is in fact only used once or twice

And I'd be surprised no one ever thought about it before, so maybe it's worth investigating if Symfony provides a service or an abstraction of some sort to ease this caching process for attribute resolving Like maybe this should be done in a compiler pass or something so that all the reflection part is done once and the listener could get the values from request attributes or somewhere else I don't know

Reply:
I agree with you, it could be cool to add a caching process for all attributes and maybe in the compiler pass. 🤔
And when we move `@AdminSecurity` into an attribute, this could be really necessary in fact.

Anyway, I set this PR into draft to think about it! Thanks about your comment 😄